### ISSUE TITLE
Proposal: Showing Next Episode 

### ISSUE THREAD

**Issue by elgatito** [scakemyer/plugin.video.quasar#703]
There is a big task for showing next episodes. The easy way is currently enabled - query each episode, start download and show it. End of an episode fires search for the next query and so on.
Problems start when providers do not found results for separate episode searches.
Here are can search for season itself, not episode. But that needs too much manual efforts from end-user, first search it launched, then select manually an episode from the list of files. When next episode expected - same actions required. 

I'm not sure about ability to store meta information for Seasons/Episodes in Kodi Library. But if it is possible then:
- Season assignment is applied to specific season and included episodes;
- Episode assignment is applied to specific episode of a season;

So:
- Each Season from Quasar browser can be assigned to a stream, which uses search season function in providers, this stream is saved for a season and episodes (with ability to re-run assignment, if season is not finished, or you don't like the selected stream, whatever);
- Each Season in Kodi Library should be linked to Quasar's item (the one, we use to click "Add to Library" action), to use same assignments when user fires 'Play episode' or 'Play season' on a Library's Item;
- If there is no Season assignment - run episode search in providers, if nothing found - run season search;
- Any stream selection should save it for future use, without asking and searching, when user wants to play same item. That should be detectable and new item coming up in context menu like "Update the stream" to rerun assignment;

If user asking to play episode, while we have a Season's assigned stream :
- If count of episodes in a library matches stream items - we can assume we have all episodes there;
- If list of files in a stream is sorted and we consider we have full season - we can think of a relation [n-th episode] = [n-th stream item]. 
- If there is "next" episode in the stream - start downloading when "current" is completed (that enables quick switch to next episode, without buffering. Probably needs a General option in Settings to enable that);
- When next episode needed - use n-th+1 item to download, using same stream.

General idea is to select Stream once for a season, and switching to the next episode should not require more actions if that is possible with season_search action by providers.

## Expected Behavior

Next episode should automatically start without asking anything if that is possible.

## Current Behavior

Next episode fires search and if not found - needs manual selecting for a user.

## Possible Solution

If providers run season_search, then use selected source to handle all episodes in the season.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used:
* Environment name and version (e.g. Kodi 15/16/17, SPMC, OSMC, OpenELEC):
* Operating System and version (Android 4/5/6, Windows 7/8/10, macOS 10.11, Ubuntu 16.10):
* Link to Kodi debug log:
<!--- See http://kodi.wiki/view/Log_file/Easy or paste to http://paste.ubuntu.com directly

**Comment by elgatito** [comment]
What will happen now if user do this:
1) user selects an episode in a Kodi library;
2) quasar fetches an item with multiple files (season pack);
3) user selects an episode, watch it to the end;
4) Kodi switches to the next;
5) Quasar asks to choose a file, but the file is already on the disk;

Will it just begin playing? 

I like the idea of downloading current episode, and when it's done, download the next one only, so switch can be done quickly. 
So 'delete file' action,  in most cases, usually apply to 'watched' items. If the item is watched - we can delete it. If not - ask to store or not (to resume later).

**Comment by nix87** [comment]
It would be even better and faster if Quasar can start playing already downloaded episode without asking anything? Right now if you want to play already downloaded episode through Quasar, he will detect that file and aks if you wanna play already matched file. That feature is one of my favorites but it can be faster if Quasar don't ask anything just play matched episode. Maybe it would be better if there is option to choose if you wanna Quasar question or not?

If that is possible we can try next step and that is making compatible Next Up service with Quasar, cause it acts basically like Netflix with all that counting before playing next episode and all. I figured that Next Up probably have problems with .strm files. Next Up works perfectly with files already saved at your storage and I did that, I dowwnloaded two episodes with Quasar and tried to use Next Up with them, but it's not working. I wondered why is that cause I already had two episodes full downloaded at mine HDD and figured it  probably cause Quasar trigger .strm files from my Kodi library and Next Up won't work with them.
